file/builder.go

```diff
@@ -266,16 +266,26 @@ func (b *stateBuilder) consumers() {
 
 		var mtlsAuths []kong.MTLSAuth
 		for _, cred := range c.MTLSAuths {
-			cred.Consumer = &kong.Consumer{
-				ID: kong.String(*c.ID),
-			}
+			cred.Consumer = c.asKongConsumer()
 			mtlsAuths = append(mtlsAuths, *cred)
 		}
 
 		b.ingestMTLSAuths(mtlsAuths)
 	}
 }
 
+func (c *FConsumer) asKongConsumer() *kong.Consumer {
+	return &kong.Consumer{ID: kong.String(*c.ID), Username: kong.String(*c.Username)}
+}
+
+func (s *FService) asKongService() *kong.Service {
+	return &kong.Service{ID: kong.String(*s.ID), Name: kong.String(*s.Name)}
+}
+
+func (r *FRoute) asKongRoute() *kong.Route {
+	return &kong.Route{ID: kong.String(*r.ID), Name: kong.String(*r.Name)}
```

Comment:
Name is not a required field for Route or Service.
Please guard dereferences with a nil check. The same applies to consumer because, at runtime, it may happen that the object may not have username (even though it is a required field).